dts/arm/atmel/samd5x.dtsi

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (c) 2019 ML!PA Consulting GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m4";
+			reg = <0>;
+		};
+	};
+
+	sram0: memory@20000000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x20000000 0x40000>;
+	};
+
+	backup0: memory@47000000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x47000000 0x2000>;
+	};
+
+	flash0: flash@0 {
+		compatible = "soc-nv-flash";
+		label = "FLASH_0";
+		write-block-size = <64>;
```

Comment:
I am less sure about this, but I think this should be 32:  `FLASH_PAGE_SIZE` is 512 for SAMD5x in the ASF headers, and with 16 pages/block = 32 bytes/page (I really wish they'd just spell this out somewhere, but I didn't find it any any obvious locations).

Reply:
Actually on further inspection: this is `FLASH_PAGE_SIZE` is 64 on SAMD2x (i.e. at least the terminology is consistent, I guess).  So this should probably be 512 (which makes the erasure unit 8KiB, which is consistent with the ASF headers).

So I think the right way to do this, is to set `ROW_SIZE` to `NVMCTRL_ROW_SIZE` or `NVMCTRL_BLOCK_SIZE` (depending on which is defined from ASF), then calculate `PAGES_PER_ROW` from `FLASH_PAGE_SIZE / ROW_SIZE`.

Unfortunately, I don't see any definition that could be used for `LOCK_REGIONS`, which looks like it's 32 on SAMD5x.

Reply:
Thank you! With this `samples/drivers/flash_shell` is mostly working.
(I can read/write and erase pages once, when I try to erase a second page, it crashes at `NVMCTRL->CTRLB.reg = NVMCTRL_CTRLB_CMD_UR | NVMCTRL_CTRLB_CMDEX_KEY;` - I'll still have to investigate this.

I've skimmed a few more data sheets too and it looks like `LOCK_REGIONS` is something for the device tree:

MCU | LOCK_REGIONS
------|------------------
SAMD2x | 16
SAML2x  | 16
SAML10  |  2
SAML11  |  4
SAMD5x  | 32